backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxValidatorApi.scala

```diff
@@ -91,6 +91,10 @@ class VeloxValidatorApi extends ValidatorApi {
   override def doColumnarShuffleExchangeExecValidate(
       outputPartitioning: Partitioning,
       child: SparkPlan): Option[String] = {
+    if (child.output.isEmpty) {
+      // See: https://github.com/apache/incubator-gluten/issues/7600.
+      return Some("Shuffle with empty schema is not supported")
+    }
```

Comment:
Empty schema batches should pass through shuffle so that they can be handled in reducer-side operators within the row number information they carried on.

cc @marin-ma

Reply:
Got it. Thanks!